### PR TITLE
fix: On connect Tab Switch error

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
To see the changes in the PR please make gitSyncReducer's `isGitSyncModalOpen` = true

## Description

Fixes #

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: hotfix/git-connect-tabSwitchError 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.73 **(-0.03)** | 36.55 **(0.01)** | 33.75 **(-0.06)** | 55.35 **(-0.02)**
 :green_circle: | app/client/src/actions/gitSyncActions.ts | 64.06 **(0.6)** | 100 **(0)** | 0 **(0)** | 75 **(4.17)**
 :red_circle: | app/client/src/api/GitSyncAPI.tsx | 54.17 **(-0.83)** | 100 **(0)** | 12.5 **(-4.17)** | 54.55 **(-1.01)**
 :green_circle: | app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx | 25.4 **(0.4)** | 18.18 **(0.94)** | 3.13 **(0)** | 28.48 **(0.5)**
 :red_circle: | app/client/src/reducers/uiReducers/gitSyncReducer.ts | 21.43 **(-5.84)** | 100 **(0)** | 0 **(0)** | 21.43 **(-5.84)**
 :red_circle: | app/client/src/sagas/GitSyncSagas.ts | 15.19 **(-4.81)** | 4.76 **(-1.12)** | 12.5 **(-4.17)** | 20 **(-5.53)**
 :red_circle: | app/client/src/selectors/gitSyncSelectors.tsx | 63.41 **(-0.69)** | 100 **(0)** | 0 **(0)** | 55.17 **(-0.39)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.71 **(0.24)** | 39.13 **(0.87)** | 36.21 **(0)** | 55.35 **(0.27)**</details>

### CODE REVIEW

File: app/client/src/entities/GitSync.ts [hunk 1]
```diff
@@ -8,3 +8,8 @@ export type GitConfig = {
   authorName: string;
   authorEmail: string;
 };
+
+export type LocalGitConfig = {
+  authorName: string;
+  authorEmail: string;
+};
```
Comment (rishabhsaxena) [inline app/client/src/entities/GitSync.ts:15]: nit: this is same as git config type above

File: app/client/src/pages/Editor/gitSync/GitSyncModal.tsx [hunk 2]
```diff
@@ -78,10 +79,13 @@ function GitSyncModal() {
...
   const setActiveTabIndex = (index: number) =>
     dispatch(setIsGitSyncModalOpen({ isOpen: true, tab: index }));
-  const isGitConnected = useIsGitConnected();
+  const gitMetaData = useSelector(getCurrentAppGitMetaData);
+  const remoteUrlInStore = gitMetaData?.remoteUrl;
   let initialTabIndex = 0;
   let menuOptions: Array<{ key: MENU_ITEM; title: string }> = [];
-  if (!isGitConnected) {
+
+  debug({ gitMetaData });
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/gitSync/GitSyncModal.tsx:87]: nit: should remove logs

File: app/client/src/pages/Editor/gitSync/Tabs/GitConnection.tsx [hunk 3]
```diff
@@ -154,19 +167,47 @@ function GitConnection({ isImport, onSuccess }: Props) {
...
+      initialAuthInfo = {
+        authorName: localGitConfig.authorName || "",
+        authorEmail: localGitConfig.authorEmail || "",
+      };
+    }
+    // TODO: fetch local config
+    return initialAuthInfo;
+  };
+
+  const initialAuthorInfoRef = useRef(getInitGitConfig());
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/gitSync/Tabs/GitConnection.tsx:203]: QQ: do we need a ref here, the state in the store should be the initial values?

File: app/client/src/reducers/uiReducers/applicationsReducer.tsx [hunk 4]
```diff
@@ -395,11 +396,12 @@ const applicationsReducer = createReducer(initialState, {
     state: ApplicationsReduxState,
     action: ReduxAction<ConnectToGitResponse>,
   ) => {
+    debug({ action });
```
Comment (rishabhsaxena) [inline app/client/src/reducers/uiReducers/applicationsReducer.tsx:399]: nit: should remove/comment logs

**Review by rishabhsaxena — APPROVED** [review]

File: app/client/src/pages/Editor/gitSync/Tabs/GitConnection.tsx (hunk 3 above)
Comment (Rishabh-Rathod) [reply app/client/src/pages/Editor/gitSync/Tabs/GitConnection.tsx:203]: Actually, `authorInfo` could be picked from the local config, global config, user profile and to identify the remote URL is changed. 

we need to know the initial value somewhere, hence I maintained this.

Comment (rishabhsaxena) [reply app/client/src/pages/Editor/gitSync/Tabs/GitConnection.tsx:203]: got it, I guess we shouldn't refer to the profile now, instead rely on local / global config?
also, on switching b/w local/global we can reset the profile values? so we could check b/w `authorName`, `authorEmail`, and the `flag` (for local vs global)?

**Review by rishabhsaxena — DISMISSED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**